src/order/rel_iso.lean

```diff
@@ -337,6 +340,42 @@ end
 @[simp] theorem of_monotone_coe [is_trichotomous α r] [is_asymm β s] (f : α → β) (H) :
   (@of_monotone _ _ r s _ _ f H : α → β) = f := rfl
 
+/-- `sum.inl` as a relation embedding. -/
+@[simps] def sum_lex_inl (r : α → α → Prop) (s : β → β → Prop) : r ↪r sum.lex r s :=
+{ to_fun := sum.inl,
+  inj' := sum.inl_injective,
+  map_rel_iff' := λ a b, sum.lex_inl_inl }
```

Comment:
Arguably we should have two results here, `r ↪r sum.lift_rel r s` and `sum.lift_rel r s ↪r sum.lex r s`

Reply:
The latter isn't be a relation embedding, but rather a subrelation. We do have a relation embedding `r ↪r sum.lift_rel r s` and `s ↪r sum.lift_rel r s` - I'll add them to the PR.